File: Data Structures & Algorithms/kth-smallest-integer-in-bst/submission-2.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    public int kthSmallest(TreeNode root, int k) {
        List<Integer> tree = new ArrayList<>();
        tree = inorder(root, tree);
        return tree.get(k-1);
    }

    public List<Integer> inorder(TreeNode root, List<Integer> tree){
        if(root != null){
            inorder(root.left, tree);
            tree.add(root.val);
            inorder(root.right, tree);
        }
        return tree;
    }
}

File: Data Structures & Algorithms/level-order-traversal-of-binary-tree/submission-2.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        if not root:
            return []
        ans = []


        q = deque()
        # For every node at the front, append their children (if present) to the queue
        # Then pop
        q.append(root)

        while q:
            le = len(q)
            level = []
            for i in range(0,le):
                node = q.popleft()
                if node.left:
                    q.append(node.left)
                if node.right:
                    q.append(node.right)
                level.append(node.val)
            ans.append(level)

        return ans
            
